src/main.rs

```diff
@@ -95,6 +102,58 @@ fn publish(request: Json<PublishRequest>, auth: TokenAuth) -> ApiResult<EmptyRes
     Ok(Json(EmptyResponse))
 }
 
+#[post(
+    "/upload_project?<forc_version>",
+    format = "application/gzip",
+    data = "<tarball>"
+)]
+async fn upload_project(
+    db: &State<Database>,
+    pinata_client: &State<PinataClientImpl>,
+    forc_version: &str,
+    mut tarball: TempFile<'_>,
+) -> ApiResult<UploadResponse> {
```

Comment:
I'm wondering if `forc_version` will need to be sanitised e.g could a request escape the working directory?

    /upload_project?v1../../../../../tmp

Reply:
Good call out, I will add a sanitization check.